src/Inkluzitron/Services/ImagesService.cs

```diff
@@ -21,73 +25,156 @@ namespace Inkluzitron.Services
 {
     public class ImagesService
     {
-        private ProfilePictureService ProfilePictureService { get; }
+        private DiscordSocketClient Client { get; }
+        private ImageCacheSettings CacheSettings { get; }
 
-        public ImagesService(ProfilePictureService profilePictureService)
+        private DirectoryInfo AvatarCache { get; }
+        private DirectoryInfo WhipCache { get; }
+        private DirectoryInfo SpankCache { get; }
+        private DirectoryInfo LoveCache { get; }
+        private DirectoryInfo PatCache { get; }
+        private DirectoryInfo BonkCache { get; }
+
+        private DirectoryInfo InitializeCache(string directoryName)
         {
-            ProfilePictureService = profilePictureService;
-            Directory.CreateDirectory("ImageCache");
+            var result = new DirectoryInfo(Path.Combine(CacheSettings.DirectoryPath, directoryName));
+            result.Create();
```

Comment:
Nehodí to exception, když to bude existovat?

Reply:
`DirectoryInfo.Create` i `DirectoryInfo.CreateSubdirectory` jsou v případě, že adresář už existuje, no-op.